### PR TITLE
Added dpi and savepdf options to comparison report for publications

### PR DESCRIPTION
**Pull request by ledm**
Colin asked for a higher resolution version of the images, so I added an option is the report maker to set image resolution in dpi, and also a flag to save images as a pdf.

### CODE REVIEW

File: bgcval2/bgcval2_make_report.py [hunk 1]
```diff
@@ -1636,7 +1636,7 @@ def newImageLocation(fn):
 
         if len(otherFilenames):
             # I think this is never happens anymore.
-            assert 0
+            #assert 0
```
Comment (ledm) [inline bgcval2/bgcval2_make_report.py:1639]:
```suggestion
```

**Comment by ledm** [comment]
I'm definitely getting my money's worth out of @valeriupredoi this week! Thanks for the hard work mate!

**Comment by valeriupredoi** [comment]
only that you're not paying me a single penny though :rofl: Lemme have a looksee in a jiffy :+1:

**Review by valeriupredoi — APPROVED** [review]
fine by me, but I'd add a default value for `dpi`, and I'd also add a check on how `dpi` is specified by the user - if it's not like eg `dpi=200` and it's `dpi=200x300` that should probably be caught before Matplotlib throws a hissy (basically `try: int(dpi) else: raose ValueError("that need be an int you dummy"`)) - up to you

**Comment by ledm** [comment]
If we leave the dpi as the default value (`None`) then the image will be saved using the user's default dpi settings, which can be changes using:
```
plt.rcParams['savefig.dpi'] = 300
```

**Comment by valeriupredoi** [comment]
okidoki :+1:

**Comment by ledm** [comment]
Added a int flag check for dpi.

File: README.md [hunk 2]
```diff
@@ -268,6 +270,12 @@ These values are:
  - `strictFileCheck`: 
    - A boolean which when True will raise an error if input model files are missing. 
    - Default is True, set to False to skip this check.
+ - `dpi`:
+   - Output images Dots per inch (dpi)
```
Comment (ledm) [inline README.md:274]:
```suggestion
   - The resolution in dots per inch (dpi) of the output image.
```

File: README.md [hunk 3]
```diff
@@ -268,6 +270,12 @@ These values are:
...
+   - dpi=100 is fine for most purposes but 300 is needed for highres posters/publications.
+ - `savepdf`:
+   - Outpout the image as a pdf in addition to the standard png.
```
Comment (ledm) [inline README.md:277]:
```suggestion
   - Output the image as a pdf in addition to the standard png. 
```

File: README.md [hunk 4]
```diff
@@ -268,6 +270,12 @@ These values are:
  - `strictFileCheck`: 
    - A boolean which when True will raise an error if input model files are missing. 
    - Default is True, set to False to skip this check.
+ - `dpi`:
+   - Output images Dots per inch (dpi)
+   - dpi=100 is fine for most purposes but 300 is needed for highres posters/publications.
+ - `savepdf`:
+   - Outpout the image as a pdf in addition to the standard png. 
+   - This doesn't replace the image in the report, but saves a pdf version of the image in the images directory.
```
Comment (ledm) [inline README.md:278]:
```suggestion
   - This doesn't replace the image in the report, but saves a pdf version of the image in the images directory.
   - The pdfs will be web-visible in the report image directory, but will not linked in the html.
   - To view a pdf from a report, simply click the image, and replace the `png` extension with `pdf` in the browser path. 
```

File: README.md [hunk 5]
```diff
@@ -268,6 +270,14 @@ These values are:
  - `strictFileCheck`: 
    - A boolean which when True will raise an error if input model files are missing. 
    - Default is True, set to False to skip this check.
+ - `dpi`:
+   - The resolution in dots per inch (dpi) of the output image.
+   - dpi=100 is fine for most purposes but 300 is needed for highres posters/publications.
```
Comment (ledm) [inline README.md:275]:
```suggestion
   - dpi=100 is fine for most purposes but 300 is needed for highres posters/publications.
   - If set to `None`, then the dpi will be set to your default value.
```

**Comment by ledm** [comment]
Done here. Going to merge if V is happy.

**Review by valeriupredoi — APPROVED** [review]
good stuff! Fire at will, bud :beer:

**Pull request merged**